components/brave_shields/browser/tracking_protection_service.cc

```diff
@@ -49,6 +65,146 @@ TrackingProtectionService::~TrackingProtectionService() {
+  base::AutoLock lock(frame_starting_site_map_lock_);
+  const RenderFrameIdKey key(render_process_id, render_frame_id);
+  auto iter = render_frame_key_to_starting_site_url.find(key);
+  if (iter != render_frame_key_to_starting_site_url.end()) {
+    return iter->second;
+  }
+  return GURL();
+}
+
+void TrackingProtectionService::ModifyRenderFrameKey(int old_render_process_id,
+                                                     int old_render_frame_id,
+                                                     int new_render_process_id,
+                                                     int new_render_frame_id) {
+  base::AutoLock lock(frame_starting_site_map_lock_);
+  const RenderFrameIdKey old_key(old_render_process_id, old_render_frame_id);
+  auto iter =
+      render_frame_key_to_starting_site_url.find(old_key);
+  if (iter != render_frame_key_to_starting_site_url.end()) {
+    const RenderFrameIdKey new_key(new_render_process_id, new_render_frame_id);
+    render_frame_key_to_starting_site_url.insert(
+        std::pair<RenderFrameIdKey, GURL>(new_key, iter->second));
+    render_frame_key_to_starting_site_url.erase(old_key);
+  }
+}
+
+void TrackingProtectionService::DeleteRenderFrameKey(int render_process_id,
+  int render_frame_id) {
+  base::AutoLock lock(frame_starting_site_map_lock_);
+  const RenderFrameIdKey key(render_process_id, render_frame_id);
+  render_frame_key_to_starting_site_url.erase(key);
+}
+
+bool TrackingProtectionService::ShouldStoreState(HostContentSettingsMap* map,
+                                                 int render_process_id,
+                                                 int render_frame_id,
+                                                 const GURL& top_origin_url,
+                                                 const GURL& origin_url) {
+  if (!TrackingProtectionHelper::IsSmartTrackingProtectionEnabled()) {
+    return true;
+  }
+
+  if (first_party_storage_trackers_.empty()) {
+    LOG(INFO) << "First party storage trackers list is empty";
+    return true;
+  }
+
+  std::string host = origin_url.host();
+
+  const GURL starting_site = GetStartingSiteForRenderFrame(render_process_id,
+    render_frame_id);
+
+  // If starting host is the current host, user-interaction has happened
+  // so we allow storage
+  if (starting_site.host() == host) {
+    return true;
+  }
+
+  bool allow_brave_shields = starting_site == GURL() ? false :
+    IsAllowContentSetting(map, starting_site, GURL(),
+      CONTENT_SETTINGS_TYPE_PLUGINS, brave_shields::kBraveShields);
+
+  if (!allow_brave_shields) {
+    return true;
+  }
+
+  bool allow_trackers = starting_site == GURL() ? true : IsAllowContentSetting(
+      map, starting_site, GURL(), CONTENT_SETTINGS_TYPE_PLUGINS,
+      brave_shields::kTrackers);
+
+  if (allow_trackers) {
+    return true;
+  }
+
+  // deny storage if host is found in the tracker list
+  return !(std::find(first_party_storage_trackers_.begin(),
```

Comment:
now you can just use `first_party_storage_trackers_.find()`, instead of a linear search